src/firebase/auth/functions/index.js

```diff
@@ -0,0 +1,148 @@
+"use strict";
+require("dotenv").config();
+
+const axios = require("axios");
+const {initializeApp} = require("firebase-admin/app");
+const {getFirestore} = require("firebase-admin/firestore");
+
+const functions = require("firebase-functions");
+const {getAuth} = require("firebase-admin/auth");
+
+initializeApp();
+const firestoreDb = getFirestore();
+
+/**
+ * @param {auth.UserRecord} _user The new user
+ */
+async function sendGoodByeMessage(_user) {
+  // TODO send user a goodbye message and request for feedback as well
+  return null;
+}
+
+/**
+ * @param {any} user The new user
+ */
+async function sendWelcomeMessage(user) {
+  try {
+    const emailAddress = user.emailAddress == null ? "" : user.emailAddress;
+    if (emailAddress === "") {
+      return null;
+    }
+
+    const displayName = user.displayName == null ? "" : user.displayName;
+    const endPoint = process.env.WELCOME_MESSAGE_ENDPOINT;
+    const body = {
+      "platform": "mobile",
+      "firstName": displayName,
+      "emailAddress": emailAddress,
+    };
+
+    // eslint-disable-next-line max-len
+    axios.post(endPoint, JSON.stringify(body), {headers: {"Content-Type": "application/json"}})
+        .then((res) => {
+          console.log(`Welcome message status code: ${res.status}`);
+        })
+        .catch((error) => {
+          console.error(error);
```

Comment:
We could also use npm's morgan to log these especially for production.

Reply:
Thanks @123MwanjeMike, shall add the appropriate logging functionality